modules/mono/editor/GodotTools/GodotTools.OpenVisualStudio/Program.cs

```diff
@@ -2,11 +2,13 @@
 using System.IO;
 using System.Runtime.InteropServices;
 using System.Runtime.InteropServices.ComTypes;
+using System.Runtime.Versioning;
 using System.Text.RegularExpressions;
 using EnvDTE;
 
 namespace GodotTools.OpenVisualStudio
 {
+    [SupportedOSPlatform("windows")]
```

Comment:
Since this project is only meant to be run on Windows, shouldn't this project target `net6.0-windows`? And would that mean we don't have to use the `SupportedOSPlatform` attribute?

Reply:
I assumed so initially, and maybe that is still the proper way to handle it. However, trying that gave me an error about how a `net6.0-windows` project couldn't be targeted properly (I'll have to double-check to see the exact wording)

Reply:
Alright, I've got a solution. I utilized the same conditional for importing the `OpenVisualStudio` the project in the first place to change the `GodotTools` TargetFramework to `net6.0-windows`. As predicted, this removed the need for the `SupportedOSPlatform` attribute

Reply:
Double-checked to make sure that everything still opened as expected in Visual Studio with this new format, and it does 👍

Reply:
Wait, why are we referencing this project in GodotTools? This is an executable project, not a library. Oh, maybe it's just so that `build_assemblies.py` builds it, but it's included in the GodotTools solution which is what we use with the build command so it should be enough.

Reply:
If it's able to be handled via `build_assemblies.py` alone, then that would simplify matters. I'll double-check if that's possible

Reply:
Alright, figured it out. The current implementation has `GodotTools.csproj` changing its output path if certain arguments are passed directly, as well as a ton of extra variables that would be a bit cumbersome to try and copy over to other project files. This is in contrast to the `GodotSharp` exports, which have a dedicated function in `build_assemblies.py` to copy over its built files. While initially outside the scope of this PR, it would make sense to update `build_assemblies.py` in order to remove the `GodotTools.csproj` hardcoded elements & executable dependency altogether

Reply:
Actually, scratch that; `Godot.NET.Sdk` has similarly hardcoded elements. This really is out of scope for this PR after all & would be better suited for a standalone refactoring of `build_assemblies.py` instead.

I'll leave this executable dependency as-is. Even if it's not technically used by GodotTools, it *is* what allows the executable to be copied over

Reply:
So I was only suggesting to remove the `ProjectReference`:

```diff
diff --git a/modules/mono/editor/GodotTools/GodotTools/GodotTools.csproj b/modules/mono/editor/GodotTools/GodotTools/GodotTools.csproj
index 56ae37b4dd..f44fa9d71a 100644
--- a/modules/mono/editor/GodotTools/GodotTools/GodotTools.csproj
+++ b/modules/mono/editor/GodotTools/GodotTools/GodotTools.csproj
@@ -49,7 +49,5 @@
     <ProjectReference Include="..\GodotTools.IdeMessaging\GodotTools.IdeMessaging.csproj" />
     <ProjectReference Include="..\GodotTools.ProjectEditor\GodotTools.ProjectEditor.csproj" />
     <ProjectReference Include="..\GodotTools.Core\GodotTools.Core.csproj" />
-    <!-- Include it if this is an SCons build targeting Windows, or if it's not an SCons build but we're on Windows -->
-    <ProjectReference Include="..\GodotTools.OpenVisualStudio\GodotTools.OpenVisualStudio.csproj" Condition=" '$(GodotPlatform)' == 'windows' Or ( '$(GodotPlatform)' == '' And '$(OS)' == 'Windows_NT' ) " />
   </ItemGroup>
 </Project>

```

But I guess this reference was why the build outputs the executable `GodotTools.OpenVisualStudio.exe` file to the `bin/GodotSharp/Tools` directory. Copying these properties to `GodotTools.OpenVisualStudio.csproj` should probably be enough:

https://github.com/godotengine/godot/blob/179dfdc8d78b5bd5377dd115af026df58308bdaf/modules/mono/editor/GodotTools/GodotTools/GodotTools.csproj#L8-L11
https://github.com/godotengine/godot/blob/179dfdc8d78b5bd5377dd115af026df58308bdaf/modules/mono/editor/GodotTools/GodotTools/GodotTools.csproj#L22-L28

Although to be honest all we really need is:

```csproj
  <PropertyGroup Condition=" Exists('$(SolutionDir)/../../../../bin/GodotSharp/Api/Debug/GodotSharp.dll') ">
    <OutputPath>$(SolutionDir)/../../../../bin/GodotSharp/Tools</OutputPath>
    <AppendTargetFrameworkToOutputPath>False</AppendTargetFrameworkToOutputPath>
  </PropertyGroup>
```

And I'm not sure we care about the `Condition`. But it does look cleaner with multiple properties like in `GodotTools.csproj`.

Are you saying this is not enough? I don't see the need to modify `build_assemblies.py` but maybe I misunderstood something.

Reply:
I was more thinking along the lines of wanting to remove any inherently hardcoded elements, but in retrospect that was a bit silly to worry about. Your condensed proposal worked, but I appended the existing conditional to cover our bases, so that now comes out as:
```csproj
  <PropertyGroup Condition="Exists('$(SolutionDir)/../../../../bin/GodotSharp/Api/Debug/GodotSharp.dll') And ('$(GodotPlatform)' == 'windows' Or ('$(GodotPlatform)' == '' And '$(OS)' == 'Windows_NT'))">
    <OutputPath>$(SolutionDir)/../../../../bin/GodotSharp/Tools</OutputPath>
    <AppendTargetFrameworkToOutputPath>False</AppendTargetFrameworkToOutputPath>
  </PropertyGroup>
```
Did a few local tests, and it works like a charm!